multinet/auth/util.py

```diff
@@ -0,0 +1,44 @@
+"""Utility functions for auth."""
+
+from typing import Any, Optional
+
+from multinet.errors import Unauthorized
+from multinet.types import Workspace
+from multinet.auth.types import UserInfo
+from multinet.user import current_user
+
+
+# NOTE: unfortunately, it is difficult to write a type signature for this
+# decorator. I've opened an issue to ask about this here:
+# https://github.com/python/mypy/issues/9032.
+def require_login(f: Any) -> Any:
+    """Decorate an API endpoint to check for a logged in user."""
+
+    def wrapper(workspace: str, *args: Any, **kwargs: Any) -> Any:
+        user = current_user()
+        if user is None:
+            raise Unauthorized("You must be logged in to create new workspaces")
+
+        return f(workspace, *args, **kwargs)
+
+    return wrapper
```

Comment:
Thinking ahead to when we implement different actions that will require logins (making tables and networks, for example), we'll probably want this to be a bit more general. Is it worth doing the work to make it general now or deferring it until we modify the other endpoints to require a log in?

Reply:
Do you mean the error message or the whole function?

The error message should be fixed now; I'll take care of that shortly.

Reply:
See 1387677bfceabc80985d05d29cea2d084ed45d74.

Reply:
Works for me